### ISSUE TITLE
The prefer-to-have-count rule does not deference expect arguments to verify their assigned value.

### ISSUE THREAD

**Issue by An631** [playwright-community/eslint-plugin-playwright#293]
The current prefer-to-have-count rule has a problem where any expect checks that take a variable argument will not check to make sure the value is not originating from a .count call. 

For example the following code would not be flagged as an error:

```ts
   const filesCount = await files.count();
   expect(filesCount).toBe(2);
```

The current code misses this case because we are not dereferencing the last assigned value of `filesCount` at the moment of verifying the expect argument. The rule should flag the error and return the following as a fix:

```ts
   const filesCount = files;
   await expect(filesCount).toHaveCount(2);
```
